### ISSUE TITLE
Routing fails when the diearea origin is non-zero

### ISSUE THREAD

**Issue by ax3ghazy** [The-OpenROAD-Project/TritonRoute#16]
Hello,

For an unconventional definition of the die area (e.g., with a positive offset of the origin), routing fails finding no tracks for certain pins.

This should be easy to replicate, and I have shared a couple of test cases privately; please let me know if you have questions.

This is possibly related to <a href="https://github.com/The-OpenROAD-Project/OpenROAD/issues/123">this</a>, but I included a test case that shows that this is actually not the only issue.

Thanks.

**Comment by eder-matheus** [comment]
Hi, @ax3ghazy.
I found that the problem occurs in the global routing stage. Your testcase with a positive offset of the origin has objects out-of-die (like I/O pins), which leads FastRoute to map these objects to the global routing grid in a wrong way. The generated guides may be disconnected and/or not properly covering the pins.

The testcase where the die area origin is zero doesn't have this issue, and when I modified the die area with negative origin (e.g.: (-2000, -2000)) also works.
